### PR TITLE
resolve several issues found by cppcheck

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

File: src/dlls/dbgshim/dbgshim.cpp [hunk 1]
```diff
@@ -1770,6 +1769,8 @@ CreateDebuggingInterfaceFromVersion2(
         goto Exit;
     }
 
+    FPCoreCLRCreateCordbObject fpCreate2 = NULL;
```
Comment (janvorli) [inline src/dlls/dbgshim/dbgshim.cpp]: What was the cppcheck complaining about w.r.t. this change?

Comment (chipitsine) [reply src/dlls/dbgshim/dbgshim.cpp]:
```
[src/dlls/dbgshim/dbgshim.cpp:1767] -> [src/dlls/dbgshim/dbgshim.cpp:1763]: (warning) Either the condition 'ppCordb==NULL' is redundant or there is possible null pointer dereference: ppCordb.
```

Comment (janvorli) [reply src/dlls/dbgshim/dbgshim.cpp]: It doesn't seem to make sense that moving the fpCreate2 definition location would fix a problem like that. Btw, it seems the fpCreate2 is completely unused, can you please delete it instead?

Comment (chipitsine) [reply src/dlls/dbgshim/dbgshim.cpp]: sure

File: src/utilcode/prettyprintsig.cpp [hunk 2]
```diff
@@ -335,15 +335,10 @@ static PCCOR_SIGNATURE PrettyPrintType(
...
-                        else    
-                        {       
-                            appendStrNumW(out, lowerBounds[i]);
-                            appendStrW(out, W("..."));
-                            if (sizes[i] != 0)  
-                                appendStrNumW(out, lowerBounds[i] + sizes[i] + 1);
-                        }       
+                        appendStrNumW(out, lowerBounds[i]);
+                        appendStrW(out, W("..."));
+                        if (sizes[i] != 0)
```
Comment (watfordgnf) [inline src/utilcode/prettyprintsig.cpp]: The enclosing if test ensures `sizes[i] != 0` here and below.

File: src/dlls/dbgshim/dbgshim.cpp (hunk 1 above)
Comment (chipitsine) [reply src/dlls/dbgshim/dbgshim.cpp]: done

File: src/utilcode/prettyprintsig.cpp (hunk 2 above)
Comment (chipitsine) [reply src/utilcode/prettyprintsig.cpp]: removed that check also

File: src/utilcode/prettyprintsig.cpp [hunk 3]
```diff
@@ -749,15 +743,10 @@ static HRESULT PrettyPrintTypeA(
...
-                        else
-                         {
-                             appendStrNumA(out, lowerBounds[i]);
-                             IfFailGo(appendStrA(out, "..."));
-                             if (sizes[i] != 0)
-                                 appendStrNumA(out, lowerBounds[i] + sizes[i] + 1);
-                         }
+                         appendStrNumA(out, lowerBounds[i]);
+                         IfFailGo(appendStrA(out, "..."));
+                         if (sizes[i] != 0)
```
Comment (jkotas) [inline src/utilcode/prettyprintsig.cpp]: Delete this condition here as well?

Comment (chipitsine) [reply src/utilcode/prettyprintsig.cpp]: sure

**Pull request merged**